fili-core/src/main/java/com/yahoo/bard/webservice/data/filterbuilders/DruidBoundFilterBuilder.java

```diff
@@ -0,0 +1,122 @@
+package com.yahoo.bard.webservice.data.filterbuilders;
+
+import static com.yahoo.bard.webservice.web.ErrorMessageFormat.FILTER_OPERATOR_INVALID;
+import static com.yahoo.bard.webservice.web.ErrorMessageFormat.TOO_MANY_DRUID_FILTERS;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionRowNotFoundException;
+import com.yahoo.bard.webservice.druid.model.filter.AndFilter;
+import com.yahoo.bard.webservice.druid.model.filter.BoundFilter;
+import com.yahoo.bard.webservice.druid.model.filter.Filter;
+import com.yahoo.bard.webservice.exception.TooManyDruidFiltersException;
+import com.yahoo.bard.webservice.web.ApiFilter;
+import com.yahoo.bard.webservice.web.DefaultFilterOperation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A DruidBoundFilterBuilder builds a Druid Bound Filter for supported dimensions, which can be sent directly to Druid.
+ * <p>
+ *     Druid Bound Filter currently supports the following API filter operations:
+ *          <li>{@link DefaultFilterOperation#gt}</li>
+ *          <li><{@link DefaultFilterOperation#lt}/li>
+ *          <li>{@link DefaultFilterOperation#gte}</li>
+ *          <li>{@link DefaultFilterOperation#lte}</li>
+ *          <li>{@link DefaultFilterOperation#between}</li>
+ * </p>
+ * <p>
+ *     So the filter:
+ *     {@code startDate|id-gt[2018-10-10]}
+ *     is translated into
+ *     {@code DruidBoundFilterBuilder(startDate,2018-10-10,null)}
+ *     which is serialized as following to druid
+ *     <pre><code>
+ *         {
+ *             "type": "bound",
+ *             "dimension": "startDate",
+ *             "lower": "2018-10-10",
+ *             "lowerStrict": false,
+ *             "upper": null ,
+ *             "upperStrict": false,
+ *             "ordering": null }
+ *         }
+ *     </code></pre>
+ * </p>
+ */
+public class DruidBoundFilterBuilder implements DruidFilterBuilder {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConjunctionDruidFilterBuilder.class);
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    private static final int DEFAULT_MAX_NUM_DRUID_FILTERS = 10000;
+    private static final int MAX_NUM_DRUID_FILTERS = SYSTEM_CONFIG.getIntProperty(
+            SYSTEM_CONFIG.getPackageVariableName("max_num_druid_filters"),
+            DEFAULT_MAX_NUM_DRUID_FILTERS
+    );
+
+    @Override
+    public Filter buildFilters(final Map<Dimension, Set<ApiFilter>> filterMap) throws DimensionRowNotFoundException {
+        LOG.trace("Building Druid Bound Filters using filter map: {}", filterMap);
+
+        // Assuming there is just going to be a single dimension for BOUND filter operations.
```

Comment:
We should bump this up to class level Javadoc and add a log warn or throw an exception if there are multiple dimensions

Reply:
No we are collecting Dimensions for other filter builders and not for DruidBoundFilter, this is based on Design

Reply:
I'm not sure why we're making this assumption instead of just looping over entries.